Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -353,42 +356,47 @@ def audit_formula_name
 
     def audit_license
       if formula.license.present?
-        non_standard_licenses = formula.license.map do |license|
-          next if license == :public_domain
-          next if @spdx_data["licenses"].any? { |spdx| spdx["licenseId"] == license }
-
-          license
-        end.compact
+        licenses, exceptions = SPDX.parse_license_expression formula.license
 
+        non_standard_licenses = licenses.reject { |license| SPDX.valid_license? license }
         if non_standard_licenses.present?
-          problem "Formula #{formula.name} contains non-standard SPDX licenses: #{non_standard_licenses}."
+          problem <<~EOS
+            Formula #{formula.name} contains non-standard SPDX licenses: #{non_standard_licenses}.
+            Check `https://spdx.org/licenses/` for a list of valid licenses.
+          EOS
         end
 
         if @strict
-          deprecated_licenses = formula.license.map do |license|
-            next if license == :public_domain
-            next if @spdx_data["licenses"].any? do |spdx|
-              spdx["licenseId"] == license && !spdx["isDeprecatedLicenseId"]
-            end
-
-            license
-          end.compact
-
+          deprecated_licenses = licenses.select do |license|
+            SPDX.valid_license?(license) && SPDX.deprecated_license?(license)
```

Comment:
```suggestion
            SPDX.deprecated_license?(license)
```

as it calls `valid_license?`

Reply:
`SPDX.deprecated_license?` returns true for an invalid license id which would mean that if the user enters an invalid license, they would get _both_ the "invalid license" error and the "deprecated license" error.

Without `SPDX.valid_license?`:
```console
$ brew audit --strict foo
foo:
  * Formula foo contains non-standard SPDX licenses: ["Apache-3.0"].
    For a list of valid licenses check: https://spdx.org/licenses/
Error: 1 problem in 1 formula detected
```

With `SPDX.valid_license?`:
```console
$ brew audit --strict foo
foo:
  * Formula foo contains non-standard SPDX licenses: ["Apache-3.0"].
    For a list of valid licenses check: https://spdx.org/licenses/
  * Formula foo contains deprecated SPDX licenses: ["Apache-3.0"].
    You may need to add `-only` or `-or-later` for GNU licenses (e.g. `GPL`, `LGPL`, `AGPL`, `GFDL`).
    Check `https://spdx.org/licenses` for a list of valid licenses.
Error: 2 problems in 1 formula detected
```

Reply:
> `SPDX.deprecated_license?` returns true for an invalid license id

It feels like it shouldn't?

Reply:
How about returning `nil` It's falsey so it won't need the `SPDX.valid_license?(license)` call but it feels better than returning `false`.